From 1d99909294defae1f47b3706fcabb4fde66063e9 Mon Sep 17 00:00:00 2001
From: Nicolin Chen <b42378@freescale.com>
Date: Tue, 2 Apr 2013 13:39:25 +0800
Subject: ENGR00256310 ASoC: imx-wm8962: Fix playback abnormally slower issue

  Previously, we didn't close FLL after playback/capture, which might cause
FLL work werid after a long time suspend.
  This patch manually disables FLL after playback/capture.

Acked-by: Wang Shengjiu <b02247@freescale.com>
Signed-off-by: Nicolin Chen <b42378@freescale.com>
---
 sound/soc/imx/imx-wm8962.c |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/sound/soc/imx/imx-wm8962.c b/sound/soc/imx/imx-wm8962.c
index 8cf4bdd..c99144f 100644
--- a/sound/soc/imx/imx-wm8962.c
+++ b/sound/soc/imx/imx-wm8962.c
@@ -214,6 +214,14 @@ static int imx_hifi_hw_free(struct snd_pcm_substream *substream)
 			pr_err("Failed to set MUTE: %d\n", ret);
 			return ret;
 		}
+
+		/* Disable FLL */
+		ret = snd_soc_dai_set_pll(codec_dai, WM8962_FLL,
+				WM8962_FLL_MCLK, 0, 0);
+		if (ret < 0) {
+			pr_err("Failed to set FLL: %d\n", ret);
+			return ret;
+		}
 	}
 	return 0;
 }
-- 
1.7.9.5

